### PR TITLE
Fix/2177/log missing local network ip

### LINKED ISSUE #2177: 🐛 BUG: Astro CLI Config `devOptions.hostname` not displaying local network IP.
### What version of `astro` are you using?

0.21.12

### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

After following the [quickstart](https://docs.astro.build/quick-start/) and setting `devOptions.hostname` to `0.0.0.0`, I see the following output in my terminal:

```bash
06:45 PM [astro] Server started                               71ms
06:45 PM [astro] Local: http://0.0.0.0:3000/
```
Shouldn't there be another line with the local network address?

Can't really do a minimal reproduction ... You can just follow the quickstart, change `devOptions.hostname` to `0.0.0.0`, and get the same output.

### Link to Minimal Reproducible Example

https://docs.astro.build/quick-start/

### PR DESCRIPTION
**Pull request by bholmesdev**
## Changes
Screen Shot 2022-03-08 at 2 07 36 PM

- Resolves #2177
- Display "localhost" for local IPs: `--hostname=127.0.0.1`, `--hostname=0.0.0.0`, no hostname provided
- Replace with `--hostname` IP for all other IPs
- Add some ✨ nice emojis ✨ and formatting tweaks to mirror Vite more closely

## Testing

Manually tested with different `--hostname` arguments.

## Docs

N/A

### CODE REVIEW

File: packages/astro/src/core/messages.ts [hunk 1]
```diff
@@ -23,8 +23,18 @@ export function reload({ url, reqTime }: { url: string; reqTime: number }): stri
...
 
 /** Display dev server host and startup time */
-export function devStart({ startupTime }: { startupTime: number }): string {
-	return `${pad(`Server started`, 44)} ${dim(`${Math.round(startupTime)}ms`)}`;
+export function devStart({ startupTime, port, localAddress, networkAddress, https, site }: { startupTime: number; port: number; localAddress: string; networkAddress: string; https: boolean; site: URL | undefined }): string {
+	const rootPath = site ? site.pathname : '/';
+	const toDisplayUrl = (hostname: string) => `${https ? 'https' : 'http'}://${hostname}:${port}${rootPath}`
+	const messages = [
+		``,
+		`${emoji('🚀', ' ')} ${magenta('astro v0.23.1')} ${dim(`started in ${Math.round(startupTime)}ms`)}`,
```
Comment (natemoo-re) [inline packages/astro/src/core/messages.ts:31]: We accidentally left this hardcoded, should pull in the actual version from `package.json`

Comment (bholmesdev) [reply packages/astro/src/core/messages.ts]: Fixed!

File: scripts/cmd/build.js [hunk 2]
```diff
@@ -28,7 +28,9 @@ export default async function build(...args) {
 		.map((f) => f.replace(/^'/, '').replace(/'$/, '')); // Needed for Windows: glob strings contain surrounding string chars??? remove these
 	let entryPoints = [].concat(...(await Promise.all(patterns.map((pattern) => glob(pattern, { filesOnly: true, absolute: true })))));
 
-	const { type = 'module', dependencies = {} } = await fs.readFile('./package.json').then((res) => JSON.parse(res.toString()));
+	const { type = 'module', version, dependencies = {} } = await fs.readFile('./package.json').then((res) => JSON.parse(res.toString()));
+	// expose PACKAGE_VERSION on process.env for CLI utils 
+	config.define = { 'process.env.PACKAGE_VERSION': JSON.stringify(version) };
```
[the remark this answers is not in the document]
Comment (bholmesdev) [reply scripts/cmd/build.js:33]: Great call!

**Review by natemoo-re — APPROVED** [review]
Nice! Looks good to merge once CI is all passing.

**Pull request merged**